code/src/components/AddItem.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components/macro';
+
+import { todos } from '../reducers/todos';
+
+export const AddItem = () => {
+	const [newItem, setNewItem] = useState('');
+	const dispatch = useDispatch();
+	const handleAddItem = () => {
+		dispatch(todos.actions.addItem(newItem));
+	};
+	return (
+		<div>
+			<TextInput
+				type="text"
+				value={newItem}
+				onChange={(e) => setNewItem(e.target.value)}
+				placeholder="What do you need to do today?"
+			/>
+			<Button
+				onClick={() => {
+					handleAddItem();
```

Comment:
Nice design on the todo app 🥳 You could add disabled attribute to the Button that is active if the testInput field is empty. This will make the user to add  todo before adding them to the list. Making it impossible to add empty todos in the list.

Reply:
👆 Preventing the user to submit empty todos needs to be fixed :)

Reply:
Indentation in all your files look a little big - change your VSC settings to 2 spaces instead of tab indentation :) Will make it easier to read!

Reply:
I been having this issue with indentation the whole bootcamp. Have it set to 2 spaces in VSCode, but it looks like this here anyway